platform-core-module/src/main/java/com/softicar/platform/core/module/file/smb/jcifs/JcifsSmbClient.java

```diff
@@ -1,39 +1,47 @@
 package com.softicar.platform.core.module.file.smb.jcifs;
 
+import com.softicar.platform.common.core.exceptions.SofticarException;
 import com.softicar.platform.core.module.file.smb.ISmbClient;
 import com.softicar.platform.core.module.file.smb.ISmbDirectory;
 import com.softicar.platform.core.module.file.smb.ISmbFile;
 import com.softicar.platform.core.module.file.smb.SmbCredentials;
-import jcifs.Config;
-import jcifs.smb.NtlmPasswordAuthentication;
+import java.util.Properties;
+import jcifs.CIFSException;
+import jcifs.config.PropertyConfiguration;
+import jcifs.context.BaseContext;
+import jcifs.smb.NtlmPasswordAuthenticator;
 
 public class JcifsSmbClient implements ISmbClient {
 
-	@Override
-	public void initialize() {
-
-		disableDfs();
-	}
+	private static final BaseContext BASE_CONTEXT = setupBaseContext();
 
 	@Override
 	public ISmbFile createFile(String url, SmbCredentials credentials) {
 
-		var authentication = new NtlmPasswordAuthentication(credentials.getDomain(), credentials.getUsername(), credentials.getPassword());
-		return new JcifsSmbFile(url, authentication);
+		var cifsContext =
+				BASE_CONTEXT.withCredentials(new NtlmPasswordAuthenticator(credentials.getDomain(), credentials.getUsername(), credentials.getPassword()));
+		return new JcifsSmbFile(url, cifsContext);
 	}
 
 	@Override
 	public ISmbDirectory createDirectory(String url, SmbCredentials credentials) {
 
-		var authentication = new NtlmPasswordAuthentication(credentials.getDomain(), credentials.getUsername(), credentials.getPassword());
-		return new JcifsSmbDirectory(url, authentication);
+		var cifsContext =
+				BASE_CONTEXT.withCredentials(new NtlmPasswordAuthenticator(credentials.getDomain(), credentials.getUsername(), credentials.getPassword()));
```

Comment:
`BASE_CONTEXT.withCredentials(new NtlmPasswordAuthenticator(credentials.getDomain(), credentials.getUsername(), credentials.getPassword()));`
should be extracted to a private method, instead of copy-and-pasting it in `createFile`/`createDirectory`